GraphWebApi/Controllers/GraphExplorerAppModeController.cs

```diff
@@ -2,60 +2,181 @@
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Http;
-using GraphExplorerAppModeService.Interfaces;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using Microsoft.Identity.Web;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using Microsoft.Graph;
+using System.Net;
+using Microsoft.Identity.Client;
+using System.Threading;
+using Microsoft.Extensions.Primitives;
+using GraphExplorerAppModeService.Services;
+using GraphExplorerAppModeService.Interfaces;
 
 namespace GraphWebApi.Controllers
 {
     [ApiController]
     public class GraphExplorerAppModeController : ControllerBase
     {
+        private readonly IGraphAppAuthProvider _graphClient;
+        private readonly ITokenAcquisition _tokenAcquisition;
+
+        public GraphExplorerAppModeController(ITokenAcquisition tokenAcquisition, IGraphAppAuthProvider graphServiceClient)
+        {
+            this._graphClient = graphServiceClient;
+            this._tokenAcquisition = tokenAcquisition;
+        }
         [Route("api/[controller]/{*all}")]
-        [Route("graphproxy/{*all}")]
+        [Route("graphproxy/{tenantId}/{*all}")]
         [HttpGet]
-        public async Task<IActionResult> GetAsync(string all, [FromHeader] string Authorization)
+        [AuthorizeForScopes(Scopes = new[] { "https://graph.microsoft.com/.default" })]
+        public async Task<IActionResult> GetAsync(string all, string tenantId)
         {
-            return await ProcessRequestAsync("GET", all, null, Authorization).ConfigureAwait(false);
+            return await this.ProcessRequestAsync("GET", all, null, tenantId).ConfigureAwait(false);
+        }
+
+        private async Task<string> GetTokenAsync(string tenantId)
+        {
+            // Acquire the access token.
+            string scopes = "https://graph.microsoft.com/.default";
+
+            return await _tokenAcquisition.GetAccessTokenForAppAsync(scopes, tenantId, null);
         }
 
+
         [Route("api/[controller]/{*all}")]
-        [Route("graphproxy/{*all}")]
+        [Route("graphproxy/{tenantId}/{*all}")]
```

Comment:
What was the reason for requiring tenantId to be in the query of every request?

Reply:
It was to have the users be able to change tenantId's as they wanted. I was following the format of [other api](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow) calls by Microsoft

Reply:
Its a good idea and I agree that this might be an approach we might wanna take to support the teams app to be making the calls, but since we planned to be taking the tenantId from the user token that gets passed from authorization with Darrell, lets stick with that for now

Reply:
Makes sense!! Didn't know that, thanks for letting me know :) 3794493386e404e5831c2b2da251da8a3d767c8d